### PR TITLE
Update change log for rc.7

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,31 @@
...
 
-# 10.0.0-rc.6 August 11, 2021
+# 10.0.0-rc.7 August 25, 2021
 
 **The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**
 
+## Features ✨ and improvements 🏁
+* Add generateId property for GeoJsonSource. ([#538](https://github.com/mapbox/mapbox-maps-android/pull/538))
+* Add default value to improve usability of FeatureState API. ([#588](https://github.com/mapbox/mapbox-maps-android/pull/588))
+* Added move layer API
```
Comment (kiryldz) [inline CHANGELOG.md:12]: Missing link to PR and also I would also mention how this API looks like.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,31 @@
...
+
+## Bug fixes 🐞
+* Fix snapshotter leak. Now calling MapSnapshotter#destroy truly frees memory. ([#569](https://github.com/mapbox/mapbox-maps-android/pull/569))
+* Remove the expression getter/setters for source properties. ([#568](https://github.com/mapbox/mapbox-maps-android/pull/568))
```
Comment (kiryldz) [inline CHANGELOG.md:16]: @pengdev please remind me if we're considering it as a breaking change.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,10 +2,31 @@
...
+* Remove strong ref dependency in snapshotter that was leading to a memory leak if Snapshotter#destroy was not called explicitly. ([#571](https://github.com/mapbox/mapbox-maps-android/pull/571))
+* Fix get annotation enum property crash ([#579](https://github.com/mapbox/mapbox-maps-android/pull/579))
+* Fix rendering issue for round line-join in line gradients
+* A fix of the layer paint property evaluation while transitioning from a data-driven value. It snaps immediately to the new value thus preventing of drawing stale data during the animation.
+* Reduced memory consumption when using raster layers by deleting CPU side tile bitmap copy after uploading to GPU texture.
+* Fix crash on Android when using tile requests delay API
+
+## Dependencies
+* Bump gl-native to rc.7.2, common to 17.1.0 ([#575](https://github.com/mapbox/mapbox-maps-android/pull/575))
```
Comment (kiryldz) [inline CHANGELOG.md:25]:
```suggestion
* Bump gl-native to v10.0.0-rc.7.2, common to 17.1.0 ([#575](https://github.com/mapbox/mapbox-maps-android/pull/575))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,10 +2,31 @@
...
+* Bump targetSDKVersion and compileSDKVersion to 30, robolectric version to 4.6.1. ([#514](https://github.com/mapbox/mapbox-maps-android/pull/514))
```
Comment (kiryldz) [inline CHANGELOG.md:26]: Wasn't this reverted?

**Review by kiryldz — COMMENTED** [review]
@Chaoba missing important change in features and improvements from gl-native: `Allow using combination of line-dasharray and line-gradient for line layer.`

File: CHANGELOG.md (hunk 4 above)
Comment (Chaoba) [reply CHANGELOG.md:26]: No, it isn't

File: CHANGELOG.md (hunk 2 above)
Comment (pengdev) [reply CHANGELOG.md:17]: Yes, I think it's a breaking change

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,10 +2,34 @@
...
+* Remove the expression getter/setters for source properties. ([#568](https://github.com/mapbox/mapbox-maps-android/pull/568))
+
+## Features ✨ and improvements 🏁
+* Add generateId property for GeoJsonSource. ([#538](https://github.com/mapbox/mapbox-maps-android/pull/538))
+* Add default value to improve usability of FeatureState API. ([#588](https://github.com/mapbox/mapbox-maps-android/pull/588))
+* Add Style#moveStyleLayer(layerId: String, layerPosition: LayerPosition?): Expected<String, None> API ([#563](https://github.com/mapbox/mapbox-maps-android/pull/563))
+* Allow using combination of line-dasharray and line-gradient for line layer.
+
+## Bug fixes 🐞
+* Fix snapshotter leak. Now calling MapSnapshotter#destroy truly frees memory. ([#569](https://github.com/mapbox/mapbox-maps-android/pull/569))
```
Comment (pengdev) [inline CHANGELOG.md:19]: I think this PR is overwritten by #571 below, right? cc @kiryldz

Comment (kiryldz) [reply CHANGELOG.md:19]: Yep, guess you're right and this line can be removed.
Thanks for catching!

**Review by kiryldz — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,10 +2,35 @@
...
+* Fix get annotation enum property crash ([#579](https://github.com/mapbox/mapbox-maps-android/pull/579))
+* Fix rendering issue for round line-join in line gradients ([#565](https://github.com/mapbox/mapbox-maps-android/pull/565))
+* A fix of the layer paint property evaluation while transitioning from a data-driven value. It snaps immediately to the new value thus preventing of drawing stale data during the animation.([#563](https://github.com/mapbox/mapbox-maps-android/pull/563))
+* Reduced memory consumption when using raster layers by deleting CPU side tile bitmap copy after uploading to GPU texture.([#563](https://github.com/mapbox/mapbox-maps-android/pull/563))
+* Fix crash on Android when using tile requests delay API([#563](https://github.com/mapbox/mapbox-maps-android/pull/563))
+
+## Dependencies
+* Bump gl-native to v10.0.0-rc.7, common to v17.0.0 ([#563](https://github.com/mapbox/mapbox-maps-android/pull/563))
+* Bump gl-native to v10.0.0-rc.7.1. ([#565](https://github.com/mapbox/mapbox-maps-android/pull/565))
```
Comment (pengdev) [inline CHANGELOG.md:28]: I think we could only keep the change that's applied to the release, meaning only keep `Bump gl-native to v10.0.0-rc.7.2, common to 17.1.0` wdyt?

**Review by pengdev — APPROVED** [review]

**Pull request merged**